crates/hir-expand/src/cfg_process.rs

```diff
@@ -0,0 +1,204 @@
+    let enabled = db.crate_graph()[loc.krate].cfg_options.check(&cfg_expr) != Some(false);
+    Some(enabled)
+}
+
+fn process_has_attrs_with_possible_comma<I: HasAttrs>(
+    items: impl Iterator<Item = I>,
+    loc: &MacroCallLoc,
+    db: &dyn ExpandDatabase,
+    remove: &mut FxHashSet<SyntaxElement>,
+) -> Option<()> {
+    for item in items {
+        let field_attrs = item.attrs();
+        'attrs: for attr in field_attrs {
+            if check_cfg_attr(&attr, loc, db).map(|enabled| !enabled).unwrap_or_default() {
+                debug!("censoring type {:?}", item.syntax());
+                remove.insert(item.syntax().clone().into());
+                // We need to remove the , as well
+                add_comma(&item, remove);
+                break 'attrs;
+            }
+
+            if let Some(enabled) = check_cfg_attr_attr(&attr, loc, db) {
+                if enabled {
+                    debug!("Removing cfg_attr tokens {:?}", attr);
+                    let meta = attr.meta()?;
+                    let removes_from_cfg_attr = remove_tokens_within_cfg_attr(meta)?;
+                    remove.extend(removes_from_cfg_attr);
+                } else {
+                    debug!("censoring type cfg_attr {:?}", item.syntax());
+                    remove.insert(attr.syntax().clone().into());
+                    continue;
+                }
+            }
+        }
+    }
+    Some(())
+}
+
+fn remove_tokens_within_cfg_attr(meta: Meta) -> Option<FxHashSet<SyntaxElement>> {
+    let mut remove: FxHashSet<SyntaxElement> = FxHashSet::default();
+    debug!("Enabling attribute {}", meta);
+    let meta_path = meta.path()?;
+    debug!("Removing {:?}", meta_path.syntax());
+    remove.insert(meta_path.syntax().clone().into());
+
+    let meta_tt = meta.token_tree()?;
+    debug!("meta_tt {}", meta_tt);
+    // Remove the left paren
+    remove.insert(meta_tt.l_paren_token()?.into());
+    let mut found_comma = false;
+    for tt in meta_tt.token_trees_and_tokens().skip(1) {
+        debug!("Checking {:?}", tt);
+        // Check if it is a subtree or a token. If it is a token check if it is a comma. If so, remove it and break.
+        match tt {
+            syntax::NodeOrToken::Node(node) => {
+                // Remove the entire subtree
+                remove.insert(node.syntax().clone().into());
+            }
+            syntax::NodeOrToken::Token(token) => {
+                if token.kind() == T![,] {
+                    found_comma = true;
+                    remove.insert(token.into());
+                    break;
+                }
+                remove.insert(token.into());
+            }
+        }
+    }
+    if !found_comma {
+        warn!("No comma found in {}", meta_tt);
+        return None;
+    }
+    // Remove the right paren
+    remove.insert(meta_tt.r_paren_token()?.into());
+    Some(remove)
+}
+fn add_comma(item: &impl AstNode, res: &mut FxHashSet<SyntaxElement>) {
```

Comment:
The name is a bit confusing, since technically this causes us to remove the comma from the input :)